### ISSUE TITLE
Depth analysis

### ISSUE THREAD

**Issue by mbhall88** [mbhall88/drprg#16]
One piece of analysis that would be nice to investigate is how results scale as read depth is reduced.

My thoughts we randomly subsample each sample to discrete depth thresholds (y-axis) and plot the sensitivity/specificity? FN/FP? some other metric? on the x-axis

If everyone is happy with this analysis, I thin start with depth thresholds
- 100
- 60
- 30
- 15
- 10
- 5

If 5 still looks okay, we could go lower still.

**Comment by mbhall88** [comment]
An outstanding question from the Teams chat

> For the depth analysis, what metric(s) do you think we should compare against the depth? I think trying to do sensitivity and specificity for each drug is going to get too messy as we'll have too many variables we're comparing against. 
> Is it okay to do sensitivity/specificity over all drugs combined? Or is there something else?

I've gone ahead and pooled sensitivity and specificity for all drugs. Here is the visualisation - I will need to debug some of these as both tbprofiler and drprg are still using minimum depth parameters that assume a decent amount of depth. This is more to check whether this is how we want to visualise these results.

## Nanopore

![image](https://user-images.githubusercontent.com/20403931/196863294-bc19d674-4cf2-437e-bbd6-e5afdcc8e015.png)

## Illumina

![image](https://user-images.githubusercontent.com/20403931/196863395-6dace0d9-9eed-4069-af5a-b0e61292a84e.png)


Any feedback?

**Comment by mbhall88** [comment]
I guess another approach could be just to pool the "interesting" drugs - i.e. first line and some of the common second line - so that the "rarer" drugs don't skew these results.

**Comment by iqbal-lab** [comment]
This looks fine, don't see why depth dependence would differ by drug,,except for drugs where indels matter. The results are unbelievably good for low depth nanopore?

**Comment by lachlancoin** [comment]
Sorry, just catching up.

I think these results look really good.

I think you should go ahead and make a big data frame with sensitivity and
specificity for all drugs at all depths tested.

We can then try a few different visualisation approaches perhaps?  E.g you
could plot the average over all drugs with an error bar. Happy to help
wrangle ggplot with this stuff if you like, its pretty easy once its in the
same data frame

**Comment by lachlancoin** [comment]
You could also 'facet'  over the different drugs in a grid, or facet over
the different depths.

**Comment by mbhall88** [comment]
> I think you should go ahead and make a big data frame with sensitivity and specificity for all drugs at all depths tested.

> You could also 'facet'  over the different drugs in a grid, or facet over the different depths.

Yeah, the data is in this form already. I'll have a play with panels. Some of the testers I did were pretty cluttered. I'll have more of a go on Monday and post some examples here

**Comment by mbhall88** [comment]
Here are some faceting examples on the nanopore data

### 2 columns

![Uploading image.png…]()


### 3 columns

![image](https://user-images.githubusercontent.com/20403931/197664779-57a5231f-5256-4043-b185-cc4bd363c4a1.png)

### 4 columns 

![image](https://user-images.githubusercontent.com/20403931/197664821-a8b7bf2a-6a8e-4f2c-aadf-5e2646e0ab4a.png)

### 5 columns

![image](https://user-images.githubusercontent.com/20403931/197664882-87c6cc71-f2c8-4f39-901f-4b922421af93.png)

**Comment by lachlancoin** [comment]
I think works, even 5 cols
